src/App.tsx

```diff
@@ -1,26 +1,303 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/Todos';
+import { TodoError } from './components/TodoError';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoSection } from './components/TodoSection';
+import { TodoHeader } from './components/TodoHeader';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+interface AppProp {
+  todoId: number;
+}
+
+export enum Filter {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
+
+export const App: React.FC<AppProp> = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [queryTodo, setQueryTodo] = useState<string>('');
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const [isUpdating, setIsUpdating] = useState<number[]>([]);
```

Comment:
looks like loading,  deletingTodoId, isUpdating can be one state.
and it can be array of todo ids which is processing on server (during update or delete multiply todos loader should be on each todo)